python/sparkdl/graph/input.py

```diff
@@ -0,0 +1,232 @@
+# Copyright 2017 Databricks, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+from __future__ import absolute_import, division, print_function
+
+import tensorflow as tf
+from tensorflow.core.protobuf import meta_graph_pb2
+
+import sparkdl.graph.utils as tfx
+
+__all__ = ["TFInputGraph"]
+
+class TFInputGraph(object):
+    """
+    An opaque serializable object containing TensorFlow graph.
+
+    [WARNING] This class should not be called by any user code.
+    """
+    def __init__(self):
+        raise NotImplementedError(
+            "Please do NOT construct TFInputGraph directly. Instead, use one of the helper functions")
+
+    @classmethod
+    def _new_obj_internal(cls):
+        # pylint: disable=attribute-defined-outside-init
+        obj = object.__new__(cls)
+        # TODO: for (de-)serialization, the class should correspond to a ProtocolBuffer definition.
+        obj.graph_def = None
+        obj.input_tensor_name_from_signature = None
+        obj.output_tensor_name_from_signature = None
+        return obj
+
+    @classmethod
+    def fromGraph(cls, graph, sess, feed_names, fetch_names):
+        """
+        Construct a TFInputGraphBuilder from a in memory tf.Graph object
```

Comment:
nit: an in-memory

Reply:
also, this returns a TFInputGraph not TFInputGraphBuilder right?

Reply:
updated